doc/source/raysgd/v2/user_guide.rst

```diff
@@ -126,42 +145,167 @@ Porting code to RaySGD
 
         TODO. Write about how to convert code to use horovod.
 
+.. TODO add BackendConfigs
 
+.. To make existing code from the previous SGD API, see :ref:`Backwards Compatibility <sgd-backwards-compatibility>`.
 
-Training on a large dataset
----------------------------
+Configurations
+--------------
 
-SGD provides native support for :ref:`Ray Datasets <datasets>`. You can pass in a Dataset to RaySGD via ``Trainer.run``.
-Underneath the hood, RaySGD will automatically shard the given dataset.
+With RaySGD, you can execute a training function (``train_func``) in a
+distributed manner by calling ``trainer.run(train_func)``. To pass arguments
+into the training function, you can expose a single ``config`` parameter:
+
+.. code-block:: diff
+
+    -def train_func():
+    +def train_func(config):
+
+Then, you can pass in the config dictionary as an argument to ``Trainer.run``:
+
+.. code-block:: diff
 
+    -trainer.run(train_func)
+    +config = {} # This should be populated.
+    +trainer.run(train_func, config=config)
+
+Putting this all together, you can run your training function with different
+configurations. As an example:
 
 .. code-block:: python
 
+    from ray.util.sgd.v2 import Trainer
+
     def train_func(config):
-        batch_size = config["worker_batch_size"]
-        data_shard = ray.sgd.get_data_shard()
-        dataloader = data_shard.to_torch(batch_size=batch_size)
+        results = []
+        for i in range(config["num_epochs"]):
+            results.append(i)
+        return results
 
-        for x, y in dataloader:
-            output = model(x)
-            ...
+    trainer = Trainer(backend="torch", num_workers=2)
+    trainer.start()
+    print(trainer.run(train_func, config={"num_epochs": 2}))
+    # [[0, 1], [0, 1]]
+    print(trainer.run(train_func, config={"num_epochs": 5}))
+    # [[0, 1, 2, 3, 4], [0, 1, 2, 3, 4]]
+    trainer.shutdown()
 
-        return model
+A primary use-case for ``config`` is to try different hyperparameters. To
+perform hyperparameter tuning with RaySGD, please refer to the
+:ref:`Ray Tune integration <tune-sgd>`.
 
-    trainer = Trainer(num_workers=8, backend="torch")
-    dataset = ray.data.read_csv("...").filter().pipeline(length=50)
+.. TODO add support for with_parameters
+
+.. _sgd-logging:
+
+Logging, Monitoring, and Callbacks
+----------------------------------
+
+Logging
+~~~~~~~
+
+Each ``Trainer`` will have a local directory created for logs, and each call
```

Comment:
To me, this section doesn't really belong in Logging. When I think of logging in an ML context, I think of logging results to Tensorboard for example, not log files. I think it might make more sense to move this to its own "Directory Structure" section outside of the "Logging, Monitoring, and Callbacks" section.

Reply:
Gotcha, forgot that Checkpoints was in a different section. Updated!